Sources/POSIX/realpath.swift

```diff
@@ -27,3 +28,34 @@ public func realpath(_ path: String) throws -> String {
     guard let rvv = String(validatingUTF8: rv!) else { throw SystemError.realpath(-1, path) }
     return rvv
 }
+
+internal let pathComponentSeparator = "/"
+
+/**
+ Resolves executable, both absolute and relative paths and referred from `PATH` environment variable and
+ return the absolute pathname.
+
+ All components in executable must exists when realpath(executable:) is called.
+*/
+public func realpath(executable: String) throws -> String {
+    // when executable is an absolute path like `/usr/bin/swift`
+    if executable.hasPrefix(pathComponentSeparator) {
+        return try realpath(argv0)
+    }
+    // when executable is a relative path like `./swift` or `bin/swift`
+    if executable.contains(pathComponentSeparator.first!) {
+        return try realpath(getcwd() + pathComponentSeparator + executable)
+    }
+    // when executable is resolved from PATH, it may be `swift` without any path component separator
+    if let paths = getenv("PATH")?.split(separator: ":") {
+        for path in paths {
+            let joinedPath = String((path.hasSuffix(pathComponentSeparator) ? path : path + "/") + executable)
```

Comment:
For consistency the `"/"` should probably be `pathComponentSeparator`.

But I think POSIX mandates `/` as the path separator anyway, so I don't know that it needs to be a separate constant.  That would also get rid of the `pathComponentSeparator.first!` above, which feels slightly awkward in that it assumes the separator consists of a single character.

Reply:
Sounds reasonable to me! 👍